@luxumbra/return of search

luxumbra opened the pull request:
I've reinstated the search and sorted the styling so it's a blends a bit more. I've not yet been able to style up the search results as I'd like but this is good to go if folks are happy with it.

Gnomeski commented:
Nice one Lux this will no doubt make plenty lives that bit easier! ;)

luxumbra commented:
@Gnomeski you're right about the icon looking small on mobile. I'll up the size a bit.

hammadj commented on docs/great-houses/house-of-NFTs.md:
```diff
@@ -36,78 +33,127 @@ When it comes to NFTs, you are only limited by your imagination!
+Both the marketplace for all other marketplaces plus the ability to create your own smart contract and storefront. Interesting feature is that all metadata (including the image) can be changed at any time by the contract owner.
 
-- **[Pixura](https://pixura.io/)** Provides a smart contract factory for artists to be able to create their own marketplace via the GUI, also offers developer tools to programmatically issue & manage tokens, run queries against your smart contracts with GraphQL, and search through token metadata with Elasticsearch.
+## [Pixura](https://pixura.io/)
```
I think you need to rebase on master since these changes to the NFT page already got merged

luxumbra replied:
> I think you need to rebase on master since these changes to the NFT page already got merged

will do. thanks.

luxumbra replied:
I've rebased and sorted conflicts.

hammadj commented on package.json:
```diff
@@ -5,7 +5,9 @@
     "swizzle": "docusaurus swizzle",
-    "deploy": "docusaurus deploy"
+    "deploy": "docusaurus deploy",
+    "serve": "serve build/",
+    "build:serve": "docusaurus build && serve build/"
```
Doesn't the start command serve the wiki?

luxumbra replied:
Yeah, it serves the wiki but the search can only work on a built site. As I wanted to style the results, I thought it'd be good to add a dev dependency for `serve` so to avoid having to deploy to vercel to check changes. Also I have been looking at the Playbooks page and wanted to try some stuff with the admin UI, which without serve, you can't use in development. It makes sense to be able to serve the build locally?

It's in dev deps so it shouldn't affect the actual build size?

luxumbra replied:
> Works better now :)
> I think that's fine, don't need to waste too much time to specific issues as the clear button, especially if this is a plugin and you don't really control the code.

Thanks for re-checking @pakokrew.  Reckon we're good to merge?

hammadj reviewed (changes requested):
There's a weird right padding added to the page when the search bar is collapsed. 

![image](https://user-images.githubusercontent.com/7143583/106026956-a4ffa180-6087-11eb-99f8-f82633ffc997.png)

Also I think it would be nice to have the "Search" text there instead of the icon only, at least on desktop
![image](https://user-images.githubusercontent.com/7143583/106027148-d8423080-6087-11eb-86ec-e4b5d1dc3533.png)

luxumbra commented:
> There's a weird right padding added to the page when the search bar is collapsed.
Good spot. I'll get it fixed.
 
> Also I think it would be nice to have the "Search" text there instead of the icon only, at least on desktop. See on the original docusaurus site: https://v2.docusaurus.io
> ![image](https://user-images.githubusercontent.com/7143583/106027148-d8423080-6087-11eb-86ec-e4b5d1dc3533.png)

Noted. I'll update the desktop styles to show the text also. :pray:

luxumbra commented:
> There's a weird right padding added to the page when the search bar is collapsed.

What device is this on @hammadj ?

hammadj commented:
> > There's a weird right padding added to the page when the search bar is collapsed.
> 
> What device is this on @hammadj ?

@luxumbra This is on Safari on MacOS. I think you just need to add `overflow: hidden` to the search bar and it will be fixed

luxumbra commented:
Hi @hammadj I've rebased master onto my branch and addressed the issue with the overflow on mobile. Hopefully this is good to go now. :pray:

pakokrew commented:
I have a problem when I click on the search field for the first time, it closes. 

Also, the icon for cleaning the search is not looking very good. 

https://user-images.githubusercontent.com/1371207/108214283-6d958980-7149-11eb-9876-9cd014a10454.mov

luxumbra commented:
Thanks for taking the time to review this @pakokrew  :pray: 

> I have a problem when I click on the search field for the first time, it closes.
I have experienced this on mobile but not on desktop. I'll see if I can resolve it. 
 
> Also, the icon for cleaning the search is not looking very good.
Agreed. I'll address this also.

luxumbra commented:
> Also, the icon for cleaning the search is not looking very good.

I have spent the last 2 hours trying to find where that little icon is set so I can override it but can I find it?! Nope! There's no pseudo elements apart from the `::before` I added to the search for the :mag: icon. I imagine it is coming from Lunr search plugin but am at a loss on how to switch it out for something more fitting. I have added some padding to the right to improve the whitespace but that's all I can do. If we want a better search, we could switch to the Algolia plugin.

> I have a problem when I click on the search field for the first time, it closes.

I have added a hover style to work around this issue on desktop and added the same input styling to the algolia input that replaces it when it has focus which in my testing, sorted the issue as much as poss without having to do something with JS
.

pakokrew commented:
Works better now :)
I think that's fine, don't need to waste too much time to specific issues as the clear button, especially if this is a plugin and you don't really control the code.
